src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Extensions/Symbols/IMethodSymbolExtensions.cs

```diff
@@ -15,6 +15,39 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions;
 
 internal static partial class IMethodSymbolExtensions
 {
+#if !ROSLYN_4_12_OR_LOWER
+    internal static IMethodSymbol? TryGetCorrespondingExtensionBlockMethod(this IMethodSymbol methodSymbol)
+    {
+        if (methodSymbol is not { IsStatic: true, IsImplicitlyDeclared: true, ContainingType.MightContainExtensionMethods: true })
+            return null;
+
+        // Having a compiler API to go from implementation method back to its corresponding extension member would be useful
+        // Tracked by https://github.com/dotnet/roslyn/issues/81686
+
+        foreach (var nestedType in methodSymbol.ContainingType.GetTypeMembers())
+        {
+            if (!nestedType.IsExtension || nestedType.ExtensionParameter is null)
+                continue;
+
+            foreach (var member in nestedType.GetMembers())
+            {
+                if (member is IMethodSymbol method)
+                {
+                    var associated = method.AssociatedExtensionImplementation;
+                    if (associated is null)
+                        continue;
```

Comment:
```suggestion
```

Reply:
Sorry, I'd misunderstood the suggestion in previous PR because of how the diff is shown in CodeFlow...